source/_posts/2015-11-16-immutable-strings-in-ruby-2-dot-3.markdown

```diff
@@ -0,0 +1,279 @@
+---
+layout: post
+title: "Immutable strings in Ruby 2.3"
+category:
+comments: true
+author:
+  name: Alexis Mas
+  email: alexis@wyeworks.com
+  twitter_handle: _axxiss
+  github_handle: axxiss
+  image: /images/team/
+  description: Software Engineer. Turning coffee into software since 2008.
+---
+
+String literals immutability is one of the new features you can find in
+[Ruby 2.3][1]. But, what does this imply?
+
+* Same value literals points to the same object
+* Better performance
+* Thread safety
+
+
+Currently string immutability is included as opt-in, being the first step of a
+multi stage rollout and if it works well it will be
+[enabled by default on Ruby 3.0][2]. So it still can end up being just an
+experiment. Time will tell.
+
+You can follow frozen string literals feature status in Ruby issue tracker
+[Feature #11473: Immutable String literal in Ruby 3][5]
+
+<!--more-->
+
+# Immutability
+
+Quoting [wikipedia][3]:
+
+> An immutable object is an object whose state cannot be modified after it is
+> created.
+
+As immutable objects state cannot be changed it doesn't make sense to copy the
+same object over and over. In fact, Ruby 2.3 will hold a unique instance for
+each string literal value used in the program. This means that 'a' and 'a'
+references the same object.
+
+This approach will help to avoid unnecessary data duplication, therefore
+unnecessary object allocation is prevented and it will imply that the program
+runs faster because it will spend less time in the Garbage Collector.
```

Comment:
`because it will spend less time, creating objects, destroying objects in garbage collection and also less time running the whole garbage collection process`